workflow/controller/operator.go

```diff
@@ -523,7 +567,7 @@ func assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatus) *wfv1.NodeStatus {
 		newDaemonStatus = &f
 		message = getPendingReason(pod)
 	case apiv1.PodSucceeded:
-		newPhase = wfv1.NodeSucceeded
+		newPhase, message = inferFailedReason(pod)
```

Comment:
this function call is confusing because of the function name, maybe something like checkForExceptions?

if and error is infered, what happens to the workflow status?

Reply:
i'll extract the exception handling logic from the existing pod failure inference logic and call it seperately

Reply:
extracting it doesnt actually make sense, but renaming the function does